### PR TITLE
Rootless container mount user token secret in the default 644 permission for Function Kubernetes runtime

### PR DESCRIPTION
**Pull request by zzzming**
### Motivation

In the rootless docker container, Function Kubernetes runtime needs to read /etc/auth/token. But currently the file permission is mounted as 400 under `root`.

### Modifications

Use the default Kubernetes secret mount permission, which allows non-root user to read the auth token.

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): no
  - The public API: no
  - The schema: no
  - The default values of configurations: no
  - The wire protocol: no
  - The rest endpoints: no
  - The admin cli options: no
  - Anything that affects deployment: no

### Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

File: pulsar-functions/runtime/src/main/java/org/apache/pulsar/functions/auth/KubernetesSecretsTokenAuthProvider.java [hunk 1]
```diff
@@ -97,14 +97,12 @@ public void configureAuthDataStatefulSet(V1StatefulSet statefulSet, Optional<Fun
...
                                 new V1SecretVolumeSource()
-                                        .secretName(getSecretName(new String(functionAuthData.get().getData())))
-                                        .defaultMode(256))));
+                                        .secretName(getSecretName(new String(functionAuthData.get().getData()))))));
 
         podSpec.getContainers().forEach(container -> container.setVolumeMounts(Collections.singletonList(
                 new V1VolumeMount()
                         .name(SECRET_NAME)
-                        .mountPath(DEFAULT_SECRET_MOUNT_DIR)
-                        .readOnly(true))));
```
Comment (eolivelli) [inline pulsar-functions/runtime/src/main/java/org/apache/pulsar/functions/auth/KubernetesSecretsTokenAuthProvider.java:107]: Why do we have to drop 'readonly' ?

Comment (zzzming) [reply pulsar-functions/runtime/src/main/java/org/apache/pulsar/functions/auth/KubernetesSecretsTokenAuthProvider.java:107]: We should keep ready only. It's added back in.

**Review by eolivelli — APPROVED** [review]
LGTM

@codelipenghui this is a blocker for 2.8.0 release

**Review by lhotari — APPROVED** [review]

**Comment by sijie** [comment]
@lhotari @eolivelli for functions change, please tag @jerrypeng @srkukarni @nlu90 @freeznet or me for reviews. We should have an eye from committers who understand the Pulsar Functions codebase better.

**Pull request merged**